services/provider/server/server.go

```diff
@@ -4,46 +4,314 @@ import (
+func (s *OCSProviderServer) createStorageConsumer(name, token string,
+	capacity resource.Quantity) (string, error) {
+	storageConsumer := newStorageConsumer(name, token, s.namespace, capacity)
+	err := s.consumerAPI.CreateStorageConsumer(storageConsumer)
+	if err != nil {
+		return "", err
+	}
+
+	existingStorageConsumer, err := s.consumerAPI.GetStorageConsumer(name, s.namespace)
+	if err != nil {
+		return "", err
+	}
+
+	return string(existingStorageConsumer.UID), nil
+}
+
+func newStorageConsumer(name, token, namespace string,
+	capacity resource.Quantity) *ocsv1alpha1.StorageConsumer {
+	return &ocsv1alpha1.StorageConsumer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Annotations: map[string]string{
+				common.TokenAnnotation: token,
+			},
+		},
+		Spec: ocsv1alpha1.StorageConsumerSpec{
+			Capacity: capacity,
+		},
+	}
+}
+
+func Start(port int, providerServer *OCSProviderServer, opts []grpc.ServerOption) {
 	lis, err := net.Listen("tcp", fmt.Sprintf("localhost:%d", port))
 	if err != nil {
 		klog.Fatalf("failed to listen: %v", err)
 	}
 
 	grpcServer := grpc.NewServer(opts...)
-	pb.RegisterOCSProviderServer(grpcServer, &ocsProviderServer{})
+	pb.RegisterOCSProviderServer(grpcServer, providerServer)
+	// Register reflection service on gRPC server.
+	reflection.Register(grpcServer)
 	err = grpcServer.Serve(lis)
 	if err != nil {
 		klog.Fatalf("failed to start gRPC server: %v", err)
 	}
 }
+
+func (s *OCSProviderServer) getJSONConnectionDetails(consumerResource *ocsv1alpha1.StorageConsumer) ([]externalSC.ExternalResource, error) {
+	var j []externalSC.ExternalResource
+
+	// Configmap with mon endpoints
+	configmap, err := s.consumerAPI.GetConfigMap(monConfigMap, s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get %s configMap. %v", monConfigMap, err)
+	}
+
+	// This is done to get address of first mon from the monConfigMap configmap
+	cmData := strings.Split(configmap.Data["data"], ",")
+	// Address of first mon
+	monAddr := cmData[0]
+	data := map[string]string{
+		"data":     monAddr,
+		"maxMonId": "0",
+		"mapping":  "{}",
+	}
+	name := monConfigMap
+	kind := "ConfigMap"
+
+	j = append(j, externalSC.ExternalResource{Name: name, Kind: kind, Data: data})
+
+	// Secret storing cluster mon.admin key, fsid and name
+	scMon, err := s.consumerAPI.GetSecret(monSecret, s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get %s secret. %v", monSecret, err)
+	}
+	data = map[string]string{
+		"fsid":         string(scMon.Data["fsid"]),
+		"mon-secret":   "mon-secret",
+		"admin-secret": "admin-secret",
+	}
+	name = monSecret
+	kind = "Secret"
+
+	j = append(j, externalSC.ExternalResource{Name: name, Kind: kind, Data: data})
+
+	// Ceph user secret
+	storageConsumer, err := s.consumerAPI.GetStorageConsumer(consumerResource.Name, s.namespace)
```

Comment:
No need to call GetStorageConsumer again.   `getJSONConnectionDetails` function already has StorageConsumer resource as an argument.